sw/device/lib/testing/mock_mmio.cc

```diff
@@ -0,0 +1,25 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/mock_mmio.h"
+
+#include "sw/device/lib/base/mmio.h"
+
+namespace mock_mmio {
+reg32_t MockDevice::Register() {
+  return reg32_from_addr(reinterpret_cast<uintptr_t>(this));
```

Comment:
May I suggest approaching this in a simpler way? IMHO forwarding mock function calls to a static class with mock methods would be simpler and wouldn't require any additional changes in `mmio.h`.

Reply:
This is actually pretty cool! This may the only way if we try to run tests in parallel or using a static instance causes a problem for any reason.

Reply:
The static version makes `MockDevice` not thread compatible, which I don't think is a good idea; it also means you can't use multiple `MockDevice`s in the same test. I think re-using reg32 as a carrier makes the dataflow clearer.